monai/data/decathlon_datalist.py

```diff
@@ -186,3 +188,53 @@ def check_missing_files(
                     missing_files.append(f)
 
     return missing_files
+
+
+def create_cross_validation_datalist(
+    datalist: List[Dict],
+    nfolds: int,
+    train_folds: Union[Sequence[int], int],
+    val_folds: Union[Sequence[int], int],
+    train_key: str = "training",
+    val_key: str = "validation",
+    filename: Optional[Union[Path, str]] = None,
+    shuffle: bool = True,
+    seed: int = 0,
+    check_missing: bool = False,
+    keys: Optional[KeysCollection] = None,
+    root_dir: Optional[str] = None,
+    allow_missing_keys: bool = False,
+):
+    """
+    Utility to create new Decathlon style datalist based on cross validation partition.
+
+    Args:
+        datalist: loaded list of dictionaries for all the items to partition.
+        nfolds: number of the kfold split.
+        train_folds: indices of folds for training part.
+        val_folds: indices of folds for validation part.
+        train_key: the key of train part in the new datalist, defaults to "training".
+        val_key: the key of validation part in the new datalist, defaults to "validation".
+        filename: if not None and ends with ".json", save the new datalist into JSON file.
+        shuffle: whether to shuffle the datalist before partition, defaults to `True`.
+        seed: if `shuffle` is True, set the random seed, defaults to `0`.
+        check_missing: whether to check all the files specified by `keys` are existing.
+        keys: if not None and check_missing_files is True, the expected keys to check in the datalist.
+        root_dir: if not None, provides the root dir for the relative file paths in `datalist`.
+        allow_missing_keys: if check_missing_files is `True`, whether allow missing keys in the datalist items.
+            if False, raise exception if missing. default to False.
+
+    """
+    if check_missing and keys is not None:
+        files = check_missing_files(datalist, keys, root_dir, allow_missing_keys)
+        if files:
+            raise ValueError(f"some files of the datalist are missing: {files}")
```

Comment:
We can always check and use a flag to switch between warning or error

Reply:
Added in the latest commit.

Thanks.